File: MarsRoverProb/src/com/marsroverprob/parser/RoverCommandsParser.java
package com.marsroverprob.parser;

import com.marsroverprob.helper.RoverHelper;
import com.marsroverprob.model.Rover;

public class RoverCommandsParser implements AbstractParser {

  @Override
  public Rover parse(Rover rover, String roverInput, String maxPos) {
    return RoverHelper.moveRover(rover, roverInput, maxPos);
  }
}

File: MarsRoverProblem_C#/Source/marsroverprob/parser/RoverCommandsParser.cs
using System;

using marsroverprob.parser.intf;
using marsroverprob.model;
using marsroverprob.helper;

namespace marsroverprob.parser.commdImpl
{
    class RoverCommandsParser : IAbstractParser
    {
          public Rover parse(Rover rover, String roverInput, String maxPos) {
            return RoverHelper.moveRover(rover, roverInput, maxPos);
        }
    }
}